packages/js/src/core/fetch-token.ts

```diff
@@ -0,0 +1,75 @@
+import { KeysToCamelCase } from '@silverhand/essentials';
+import { string, number, assert, type, StructError, Infer } from 'superstruct';
+import { Except } from 'type-fest';
+
+import { LogtoError } from '../utils/errors';
+import { createRequester } from '../utils/requester';
+
+export interface FetchTokenByRefreshTokenParameters {
+  clientId: string;
+  tokenEndPoint: string;
+  refreshToken: string;
+  resource?: string;
+  scope?: string[];
+}
+
+const TokenResponseSchema = type({
+  access_token: string(),
+  refresh_token: string(),
+  id_token: string(),
+  scope: string(),
+  expires_in: number(),
+});
+
+export type RefreshTokenTokenResponse = Except<
+  KeysToCamelCase<Infer<typeof TokenResponseSchema>>,
+  'scope'
+> & {
+  scope: string[];
+};
+
+export const fetchTokenByRefreshToken = async ({
+  clientId,
+  tokenEndPoint,
+  refreshToken,
+  resource,
+  scope,
+}: FetchTokenByRefreshTokenParameters): Promise<RefreshTokenTokenResponse> => {
+  const parameters = new URLSearchParams();
+  parameters.append('client_id', clientId);
+  parameters.append('refresh_token', refreshToken);
+  if (resource) {
+    parameters.append('resource', resource);
+  }
+
+  if (scope?.length) {
+    parameters.append('scope', scope.join(' '));
+  }
+
+  const requester = createRequester();
+  const response = await requester(tokenEndPoint, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: parameters,
+  });
+
+  try {
+    assert(response, TokenResponseSchema);
+    const { access_token, refresh_token, id_token, scope, expires_in } = response;
```

Comment:
nit: `{ access_token: accessToken, refresh_token: refreshToken ... } = response; `

Reply:
I tried, but personally don't quite like the outcome. For me it does NOT seem to increase readability.

![image](https://user-images.githubusercontent.com/12833674/151370628-9eac2a4a-9498-4bcb-addc-88e1320b2fa2.png)